lib/sonarlint.js

```diff
@@ -1,164 +1,390 @@
+          args.push(toUrl(path.resolve(pluginsHome, sonarphp.filename)));
+          args.push(toUrl(path.resolve(pluginsHome, sonarpython.filename)));
+          childProcess = this.spawnServer(serverHome, command, args);
+        });
+      });
     });
+  }
 
-    process.on('close', (code) => {
-      this.logger.debug(`SonarLint Language Server process exited with code ${code}`);
+  spawnServer(serverHome, command, args) {
+    this.logger.debug(`starting "${command} ${args.join(' ')}"`);
+    const childProcess = cp.spawn(command, args, { cwd: serverHome });
+    childProcess.stdout.setEncoding('utf8');
+    childProcess.stdout.on('data', (chunk) => {
+      const chunkStr = chunk.toString();
+      chunkStr.split('\n').filter((l) => l).forEach((line) => this.logger.info(`stdout ${line}`));
     });
+    this.captureServerErrors(childProcess, null);
+    childProcess.on('exit', exitCode => {
+      if (!childProcess.killed) {
+        atom.notifications.addError('SonarLint language server stopped unexpectedly.', {
+          dismissable: true,
+          description: this.processStdErr ? `<code>${this.processStdErr}</code>` : `Exit code ${exitCode}`
+        });
+      }
+      this.updateStatusBar('Stopped');
+    });
+    return childProcess;
+  }
 
-    return process;
+  checkJavaVersion (command) {
+    return new Promise((resolve, reject) => {
+      const childProcess = cp.spawn(command, [ '-showversion', '-version' ]);
+      childProcess.on('error', err => {
+        this.showJavaRequirements(
+          'SonarLint could not launch your Java runtime.',
+          err.code == 'ENOENT' ? `No Java runtime found at <b>${command}</b>.` : `Could not spawn the Java runtime <b>${command}</b>.`
+        );
+        reject();
+      });
+      let stdErr = '', stdOut = '';
+      childProcess.stderr.on('data', chunk => stdErr += chunk.toString());
+      childProcess.stdout.on('data', chunk => stdOut += chunk.toString());
+      childProcess.on('exit', exitCode => {
+        const output = stdErr + '\n' + stdOut;
+        if (exitCode === 0 && output.length > 2) {
+          const version = this.getJavaVersionFromOutput(output);
+          if (version == null) {
+            this.showJavaRequirements(
+              `SonarLint requires Java ${minJavaRuntime} but could not determine your Java version.`,
+              `Could not parse the Java '--showVersion' output <pre>${output}</pre>.`
+            );
+            reject();
+          }
+          if (version >= minJavaRuntime) {
+            this.logger.debug(`Using Java ${version} from ${command}`);
+            resolve(version);
+          } else {
+            this.showJavaRequirements(
+              `SonarLint requires Java ${minJavaRuntime} or later but found ${version}`,
+              `If you have Java ${minJavaRuntime} installed please Set Java Path correctly. If you do not please Download Java ${minJavaRuntime} or later and install it.`
```

Comment:
Why are the words "Set", "Path" and "Download" capitalized? (shouldn't these be simply lowercase?)